xapian-glib/xapian-term-iterator.cc

```diff
@@ -0,0 +1,321 @@
+/* Copyright 2017  Olly Betts
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <config.h>
+
+#include "xapian-term-iterator.h"
+#include "xapian-error-private.h"
+
+#include <xapian/iterator.h>
+
+/**
+ * SECTION:xapian-term-iterator
+ * @Title: XapianTermIterator
+ * @short_description: Iterate over terms
+ *
+ * #XapianTermIterator is a class that allows iterating over terms.
+ *
+ * A typical example is:
+ *
+ * |[<!-- language="C" -->
+ *   XapianTermIterator *iter = xapian_database_allterms (db);
+ *
+ *   while (xapian_term_iterator_next (iter))
+ *     g_print ("%s", xapian_term_iterator_get_term (iter));
+ * ]|
+ */
+
+class IteratorData {
+  public:
+    IteratorData (const IteratorData &aData) {
```

Comment:
Right, though it's actually just what you get by default, so removing it won't change much.  I'll suppress the default copy ctor and assignment operator, as we shouldn't be copying these objects.

I'll remove the equivalent unused copy ctor from the mset iterator code too.